<launch>

  <param name="/use_sim_time" value="true"/>

  <!-- Launch stage world -->
  <include file="$(find stage_worlds)/launch/laser_world.launch" />
	

  <!-- Wall follow behavior -->
  <node name="wall_follow" pkg="stage_behavior" type="wall_following_behavior" launch-prefix="xterm -e">
    <remap from="scan" to="base_scan"/>
    <remap from="cmd_vel" to="cmd_vel_1"/>
    <param name="publish_rate" value="20"/>
    <param name="bump_distance" value="1.0" />
    <param name="robot_size" value="0.33" />
  </node>

  <!-- Other behavior -->

  <node name="goal_seek" pkg="stage_behavior" type="goal_seek" launch-prefix="xterm -e">
    <remap from="cmd_vel" to="cmd_vel"/>
    <remap from="scan" to="base_scan"/>
    <param name="publish_rate" value="10"/>
  </node>
  
  
  <node name="static_goal_publisher" pkg="rostopic" type="rostopic"  
  args="pub /move_base_simple/goal geometry_msgs/PoseStamped 
  '{header: {stamp: now, frame_id: &quot; map &quot;}, pose: {position: {x: 3.25, y: -5.5, z: 0.0}, orientation: {w: 1.0}}}'"/>
 
</launch>


